charactersheet/charactersheet/models/magic_item.js

```diff
@@ -0,0 +1,86 @@
+"use strict";
+
+function MagicItem() {
+    var self = this;
+    self.ps = PersistenceService.register(MagicItem, self);
+
+    self.characterId = ko.observable(null);
+    self.magicItemName = ko.observable('');
+    self.magicItemType = ko.observableArray('');
+    self.magicItemRarity = ko.observableArray('');
+    self.magicItemRequiresAttunement = ko.observable(false);
+    self.magicItemAttuned = ko.observable(false);
+    self.magicItemMaxCharges = ko.observable(0);
+    self.magicItemCharges = ko.observable(0);
+    self.magicItemWeight = ko.observable(0);
+    self.magicItemDescription = ko.observable('');
+    self.magicItemTypeOptions = ko.observableArray(
+        ['Armor', 'Sword', 'Rod', 'Ring', 'Staff',
+         'Wand', 'Potion', 'Wondrous Item']);
+    self.magicItemRarityOptions = ko.observableArray(
+        ['Uncommon', 'Common', 'Rare', 'Rarity Varies',
+         'Very Rare', 'Legendary']);
+
+    self.chargesDisplay = ko.pureComputed(function(){
+        if(self.magicItemMaxCharges() == 0){
+            return "N/A"
+        }
+        else {
+            return self.magicItemCharges()
+        }
+    });
+
+    self.clear = function() {
```

Comment:
Could you use the KO mapping plugin [like the Profile module does](https://github.com/adventurerscodex/adventurerscodex.github.io/pull/463/files#diff-b6ec3a5e8159340c26393da329d95e86R41)? 
See PR #463